staging/src/k8s.io/client-go/kubernetes/typed/core/v1/fake/fake_pod_expansion.go

```diff
@@ -89,7 +89,7 @@ func (c *FakePods) EvictV1(ctx context.Context, eviction *policyv1.Eviction) err
 	action.Namespace = c.ns
 	action.Resource = podsResource
 	action.Subresource = "eviction"
-	action.Object = eviction
+	action.Object = &metav1.DeleteOptions{}
```

Comment:
if you're trying to model the effect an eviction has on the pod object, creating an eviction produces a *delete* operation against the pod resource, not a create request with DeleteOptions against the pods/eviction subresource

that said, thinking about this more, changing what this invokes to a pod deletion will break anyone using the fake client and expecting to observe the eviction creation

I think your first instinct to adjust the object tracker handler here was probably more correct: https://github.com/kubernetes/kubernetes/blob/80c19baa1104bd9da0e429a022af0d16bdeeb0c9/staging/src/k8s.io/client-go/testing/fixture.go#L104-L109

but instead of baking in specific knowledge of specific types, this can be improved generically by refusing to change the type of an existing object via a subresource create request

Reply:
@liggitt Forgive me for not understanding what you mean, do you mean that when action is CreateActionImpl and if action.GetObject() is Eviction, then this problem can be solved by not executing Update?

Reply:
> do you mean that when action is CreateActionImpl and if action.GetObject() is Eviction, then this problem can be solved by not executing tracker.Update?

more like when action is CreateActionImpl and type(action.GetObject()) != type(existing object), don't execute tracker.Update

Reply:
@liggitt Thanks for the hint, I made changes based on the results of our discussion, I hope this time  can pass the Review requested